pkg/collection/entry/entry.go

```diff
@@ -0,0 +1,63 @@
+// Copyright 2020 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package entry
+
+import (
+	"errors"
+
+	"github.com/ethersphere/bee/pkg/collection"
+	"github.com/ethersphere/bee/pkg/swarm"
+)
+
+var (
+	serializedDataSize = swarm.SectionSize * 2
+)
+
+// Entry provides addition of metadata to a data reference.
+// Implements collection.Entry.
+type Entry struct {
+	reference swarm.Address
+	metadata  swarm.Address
+}
+
+// New creates a new Entry.
+func New(reference swarm.Address) *Entry {
+	return &Entry{
+		reference: reference,
+	}
+}
+
+// SetMetadata sets the metadata for the data reference in Entry.
+func (e *Entry) SetMetadata(metadataAddress swarm.Address) {
```

Comment:
i wonder, why would you like to separate this? why doesn't the constructor take a reference _and_ a metadata address/entry

Reply:
Right, this method is not needed. Metadata address can be set in the constructor.